### PR TITLE
misc(events): update validations when creating batch events

### PR DESCRIPTION
**Pull request by lovrocolic**
## Context

Update validations when creating batch events

## Description

We should add more detail validation response to the user when creating events in batch. Currently we have set of validations that is performed during the Job processing, but we also want to add the same scope of validations to the user (in API response)

### CODE REVIEW

File: app/services/events/create_batch_service.rb [hunk 1]
```diff
@@ -2,21 +2,20 @@
...
-    ALL_REQUIRED_PARAMS = %i[transaction_id code external_subscription_ids].freeze
-
-    def validate_params(params:)
-      params_errors = ALL_REQUIRED_PARAMS.each_with_object({}) do |key, errors|
-        errors[key] = ['value_is_mandatory'] if params[key].blank?
-      end
-      return result if params_errors.blank?
+    def validate_params(organization:, params:)
+      Events::ValidateCreationService.call(
+        organization: organization,
```
Comment (jdenquin) [inline app/services/events/create_batch_service.rb:7]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: app/services/events/create_batch_service.rb [hunk 2]
```diff
@@ -2,21 +2,20 @@
...
+        params: params,
```
Comment (jdenquin) [inline app/services/events/create_batch_service.rb:8]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: app/services/events/create_batch_service.rb [hunk 3]
```diff
@@ -2,21 +2,20 @@
...
+        customer: customer(organization:, params:),
+        result: result,
```
Comment (jdenquin) [inline app/services/events/create_batch_service.rb:10]: Style/HashSyntax: Omit the hash value. (https://rubystyle.guide#hash-literals)

File: spec/services/events/create_batch_service_spec.rb [hunk 4]
```diff
@@ -8,39 +8,107 @@
...
+        end
+      end
+    end
+
+    context 'with invalid customer reference' do
+      let(:subscription3) { create(:active_subscription) }
+      let(:event_arguments) do
+        {
+          transaction_id: SecureRandom.uuid,
+          external_subscription_ids: [subscription3.external_id, subscription2.external_id],
```
Comment (jdenquin) [inline spec/services/events/create_batch_service_spec.rb:78]: RSpec/ScatteredLet: Group all let/let! blocks in the example group together. (https://www.rubydoc.info/gems/rubocop-rspec/RuboCop/Cop/RSpec/ScatteredLet)

**Comment by rsempe** [comment]
Can we add the `invalid_properties_error` as well for batch events?

**Review by rsempe — APPROVED** [review]

**Review by vincent-pochet — APPROVED** [review]

**Pull request merged**